test/integration/dataContract/validateDataContractFactory.spec.js

```diff
@@ -1102,10 +1102,53 @@ describe('validateDataContractFactory', () => {
         const [error] = result.getErrors();
 
         expect(error.dataPath).to.equal(
-          '.documents[\'withContentEncoding\'].properties[\'binaryField\'].contentEncoding',
+          '.documents[\'withContentEncoding\'].properties[\'base64Field\'].contentEncoding',
+        );
+        expect(error.keyword).to.equal('enum');
+      });
+
+      it('should have `pattern` set if `contentEncoding` is set to `base58`', async () => {
+        delete rawDataContract.documents.withContentEncoding.properties.base58Field.pattern;
+
+        const result = await validateDataContract(rawDataContract);
+
+        expectJsonSchemaError(result);
+
+        const [error] = result.getErrors();
+
+        expect(error.dataPath).to.equal('.documents[\'withContentEncoding\'].properties[\'base58Field\']');
+        expect(error.keyword).to.equal('required');
+      });
+
+      it('should have specific `pattern` set if `contentEncoding` is set to `base58`', async () => {
+        rawDataContract.documents.withContentEncoding.properties.base58Field.pattern = '.*';
+
+        const result = await validateDataContract(rawDataContract);
+
+        expectJsonSchemaError(result);
+
+        const [error] = result.getErrors();
+
+        expect(error.dataPath).to.equal(
+          '.documents[\'withContentEncoding\'].properties[\'base58Field\'].pattern',
         );
         expect(error.keyword).to.equal('const');
       });
+
+      it('should have `contentEncoding` set to `base58` if former has been set', async () => {
```

Comment:
This test duplicates "should have `contentEncoding` set to `base64` if former has been set". I would just rename the first one: "should have `contentEncoding` set to either `base64` or `base58`"